### PR TITLE
April 14th (Release) blog

### PR DESCRIPTION
**Pull request by adrianhall**

### CODE REVIEW

File: _posts/2020-04-14-april-release.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+## In depth: Where should you store secrets?
+
+It's common to hear the advice "Don't store secrets in source code."  This is excellent advice.  Our source code is increasingly stored in source code control systems that are hosted in the cloud (like GitHub or Azure DevOps), but where is a good place to store these secrets and what form should they take?  We've got some best practices and some code snippets for you this week to get you started right.
+
+### Pick the right authentication scheme
+
+You should always secure each resource individually.  All compute services in Azure support a "managed identity".  This provides you with a service principal within Azure Active Directory that you can use to call other services.  The `DefaultAzureCredential` uses managed identities out of the box, so this is an excellent way to get started.  You have to specify which permissions the managed identity has within Azure Active Directory.  This is normally as simple as giving the managed identity the right roles so that they can access the resources needed.
```
Comment (jongio) [inline _posts/2020-04-14-april-release.md:46]: Is this true that **ALL* compute service support MI?

File: _posts/2020-04-14-april-release.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Some services only support username/password type authentication; yet others only support API keys.  These are great for getting started, but suffer because you have to remember to rotate those keys on a regular basis.  Centralizing authentication in a place designed for it allows for policy enforcement, and ensures you know which passwords need rotating.
+
+### Avoid connection strings
+
+When you are developing service applications, you want them to be long-running.  If you have good password hygiene, then you are rotating the passwords to all your cloud resources on a regular basis.  At some point, the password you are using in your application (that is embedded in a connection string) is going to change, causing you to restart your application.  Of course, you've designed your application reith resiliency in mind, so this involves modifying the password wherever you store them and restarting your app, but you can avoid that restart.
```
Comment (jongio) [inline _posts/2020-04-14-april-release.md:52]: with instead of reith

File: _posts/2020-04-14-april-release.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+You should instead use the `DefaultAzureCredential` to authenticate to services that can support it.  The `DefaultAzureCredential` automatically deals with changing passwords.  You can rotate keys on a regular basis without restarting your service. For more information on the `DefaultAzureCredential`, see [our blog post]({% post_url 2020-02-25-defaultazurecredentials %}).
```
Comment (jongio) [inline _posts/2020-04-14-april-release.md:54]: technically this is ManagedIdentityCredential

File: _posts/2020-04-14-april-release.md (hunk 1 above)
Comment (adrianhall) [reply _posts/2020-04-14-april-release.md:46]: Good point - Batch and Spring Cloud don't as yet - I'd forgotten Spring Cloud had been added.

File: _posts/2020-04-14-april-release.md (hunk 3 above)
Comment (adrianhall) [reply _posts/2020-04-14-april-release.md:54]: Will make clearer.

**Pull request merged**